### DISCUSSION TITLE
Could not get record changes in subscribe

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2325] [Q&A]
I am using custom models. When update record like below, could not get record changes in frontend. 

```
article, err := FindArticleBySlug(app.Dao(), "lorem-ipsum")
if err != nil {
    return err
}

article.Description = "Hello World!"

if err := app.Dao().Save(article); err != nil {
    return err
}
```

It's working fine in `form.Submit()` and `app.Dao().SaveRecord(record)`.

**Comment by ganigeorgiev** [comment]
As noted in https://pocketbase.io/docs/custom-models/#working-with-the-custom-model-struct, custom model structs at the moment don't work with the `models.Record` related hooks (which are also used by the realtime apis).

**Reply by Oggy4th** [reply]
So how to listen changes when using custom models?

**Reply by ganigeorgiev** [reply]
Unfortunately at the moment there is no easy way to do it.

_I've added it to the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=26152255) to consider it for the next release by internally converting the custom struct to `Record` (aka. constructing and fetching the related `Collection` and `Record` models based on the `model.TableName()` and `model.GetId()` `models.Model` interface fields)._

**Reply by Oggy4th** [reply]
Should I go back to `form.Submit()` or `app.Dao().SaveRecord(record)` to listen change?

**Reply by ganigeorgiev** [reply]
Yes, at the moment there is no other easier way.

I'll try to experiment with the auto internal conversion approach described above sometime later today and eventually may push the changes as part of v0.15.1.

**Reply by Oggy4th** [reply]
My whole code used custom models. Should I go back or should I wait v0.15.1, Is there any ETA for v0.15.1

**Reply by ganigeorgiev** [reply]
I'll try to experiment with a fix today, but there are no hard ETAs.

Please note that while the above described workaround may work with the realtime events, the other `Record` related hooks still will work only with the `models.Record` struct.

**Reply by ganigeorgiev** [reply]
@Oggy4th I've pushed a fix in `master`. You can run `go get github.com/pocketbase/pocketbase@master` to update your `go.mod` and test it.

Final v0.15.1 release will be available sometime later this week.

**Reply by dschissler** [reply]
@Oggy4th Early in my project I abandoned using custom models due to endless woes.  There are lots of edge cases, and I'm not convinced that the two patterns will be compatible anytime soon.

**Reply by pnmcosta** [reply]
> Please note that while the above described workaround may work with the realtime events, the other `Record` related hooks still will work only with the `models.Record` struct.

Just for clarification, does this mean that even with v0.15.1+, [crud events](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks) are not raised if saving custom models?

Thanks

**Reply by ganigeorgiev** [reply]
@pnmcosta The linked `OnRecord*Request` hooks fire only when performing the related **web API request** (aka. list/view/create/update/delete of a record). The related hook events operate only with `e.Record`, aka. there are no custom models here.

If that doesn't help, please file a new discussion with more details and a code sample what you are trying to do.

**Reply by pnmcosta** [reply]
Thanks for the clarification, yes I'll create a new discussion, thank you.
